iob_soc.py

```diff
@@ -40,308 +43,606 @@ class iob_soc(iob_module):
+            min="1",
+            max="32",
+            descr="AXI ID bus width",
+        ),
+        iob_conf(
+            name="AXI_ADDR_W",
+            type="F",
+            val="`IOB_SOC_MEM_ADDR_W",
+            min="1",
+            max="32",
+            descr="AXI address bus width",
+        ),
+        iob_conf(
+            name="AXI_DATA_W",
+            type="F",
+            val="`IOB_SOC_DATA_W",
+            min="1",
+            max="32",
+            descr="AXI data bus width",
+        ),
+        iob_conf(
+            name="AXI_LEN_W",
+            type="F",
+            val="4",
+            min="1",
+            max="4",
+            descr="AXI burst length width",
+        ),
+        iob_conf(
+            name="MEM_ADDR_OFFSET",
+            type="F",
+            val="0",
+            min="0",
+            max="NA",
+            descr="Offset of memory address",
+        ),
     ]
-    ios = [
-        {
-            "name": "clk_en_rst",
-            "type": "slave",
-            "port_prefix": "",
-            "wire_prefix": "",
-            "descr": "Clock, enable, and reset",
-            "ports": [],
+
+    # Create wire for connection with IOs
+    clk_en_rst = iob_interface("clk_en_rst")
+    clk = clk_en_rst.wires.clk
+    arst = clk_en_rst.wires.arst
+    cke = clk_en_rst.wires.cke
+
+    trap = iob_wire("trap", width=1)
+
+    axi = iob_interface(
+        name="axi",
+        type="master",
+        wire_prefix="",
+        port_prefix="",
+        mult="",  # Will be filled automatically
+        widths={
+            "ID_W": "AXI_ID_W",
+            "ADDR_W": "AXI_ADDR_W",
+            "DATA_W": "AXI_DATA_W",
+            "LEN_W": "AXI_LEN_W",
         },
-        {
-            "name": "trap",
-            "type": "master",
-            "port_prefix": "",
-            "wire_prefix": "",
-            "descr": "iob-soc trap signal",
-            "ports": [
-                {
-                    "name": "trap",
-                    "direction": "output",
-                    "width": 1,
-                    "descr": "CPU trap signal",
-                },
+        )
+
+    txd = iob_wire(name='txd', width=1)
+    rxd = iob_wire(name='rxd', width=1)
+    cts = iob_wire(name='cts', width=1)
+    rts = iob_wire(name='rts', width=1)
+
+    ios = [
+        iob_interface(
+            name="clk_en_rst",
+            type="slave",
+            port_prefix="",
+            wire_prefix="",
+            descr="Clock, enable, and reset",
+            ports=[],
+            connect_to=clk_en_rst,
+        ),
+        iob_interface(
+            name="trap",
+            type="master",
+            port_prefix="",
+            wire_prefix="",
+            descr="iob-soc trap signal",
+            ports=[
+                iob_port(
+                    name="trap",
+                    direction="output",
+                    width=1,
+                    descr="CPU trap signal",
+                    connect_to=trap
```

Comment:
We can connect a single port to a wire (created previously) using the `connect_to` attribute.

Reply:
Removed this attribute in commit 86b8840a9b5e377a9ae2ead2a4d6127883c0a874